### PR TITLE
WT-7929 Investigate a solution to avoid FTDC stalls during checkpoint

### PR DESCRIPTION
**Pull request by mm-ng**
@agorrod @etienneptl Let me know if this approach is along the lines of what you've been thinking to simplify the statistics retrieval flow.

### CODE REVIEW

File: src/schema/schema_stat.c
Comment (agorrod) [inline src/schema/schema_stat.c:160]: I think this could be simplified, you could do something like this:

```
uri = table->cgroups[0]->name;
if (!WT_PREFIX_SKIP(uri, "file:"))
            return (__wt_unexpected_object_type(session, uri, "file:"));
ret = __wt_curstat_file_init(session, uri, NULL, cfg, CST);
```

Rather than creating a `statistics:file...` URI, and then stripping the `statistics:` off again in the file sub-call.

**Review by agorrod — COMMENTED** [review]
@mm-ng  - this looks good to me. Did you (or could you) check whether MongoDB takes the `is_simple` path? I'm keen to ensure this improves the user experience for MongoDB.

**Comment by mm-ng** [comment]
> @mm-ng - this looks good to me. Did you (or could you) check whether MongoDB takes the `is_simple` path? I'm keen to ensure this improves the user experience for MongoDB.

I definitely plan to check this through some form of testing (most likely using many-collections test).

File: src/schema/schema_stat.c
Comment (mm-ng) [reply src/schema/schema_stat.c:160]: @agorrod I'm not sure if this approach will work - I think the uris can start with `colgroup:`, which wouldn't match with `file:`

File: src/schema/schema_stat.c
Comment (etienneptl) [inline src/schema/schema_stat.c:162]: I would prefer having `goto done` rather than `goto err` as this path should not be interpreted as an error by any reader. A comment could be enough too instead of adding another label.

**Comment by mm-ng** [comment]
> > @mm-ng - this looks good to me. Did you (or could you) check whether MongoDB takes the `is_simple` path? I'm keen to ensure this improves the user experience for MongoDB.
> 
> I definitely plan to check this through some form of testing (most likely using many-collections test).

It looks like MongoDB is taking the `is_simple` path, as I can see that the stat collection time is lower with these code changes - see the Jira comment in [WT-7929](https://jira.mongodb.org/browse/WT-7929).

File: src/schema/schema_stat.c
Comment (mm-ng) [reply src/schema/schema_stat.c:162]: I will add a comment here.

**Review by etienneptl — APPROVED** [review]
LGTM modulo the missing comment about the `err` label which does not need another review.

File: src/schema/schema_stat.c
Comment (sulabhM) [reply src/schema/schema_stat.c:167]: Ah, it is better to have multiple labels, in this case, "done" or "skip" along with "err". An example will be:
https://github.com/wiredtiger/wiredtiger/blob/develop/src/block/block_ext.c#L1420

**Review by sulabhM — CHANGES_REQUESTED** [review]
Small change requested to add a labele along with "err" to jump to when skipping in case of simple tables.

File: src/schema/schema_stat.c
Comment (mm-ng) [reply src/schema/schema_stat.c:167]: Thanks, I'll follow this

File: src/schema/schema_stat.c
Comment (etienneptl) [inline src/schema/schema_stat.c:162]: If we have a `goto done` here now, I suggest the following to avoid any issue in the long run:
```suggestion
        WT_ERR(__wt_curstat_init(session, buf->data, NULL, cfg, CST));
        goto done;
```
What do you think @mm-ng @sulabhM?

Comment (mm-ng) [reply src/schema/schema_stat.c:162]: Yes, I'd be happy to put this change in.

Comment (sulabhM) [reply src/schema/schema_stat.c:162]: Sounds good, resolving the conversation to avoid confusion.

**Review by sulabhM — APPROVED** [review]

File: src/schema/schema_stat.c
Comment (sulabhM) [inline src/schema/schema_stat.c:161]: CST --> cst

**Review by sulabhM — APPROVED** [review]

**Comment by mm-ng** [comment]
evergreen merge

**Pull request merged**
